Docs/developer_guide/parameter_nodes/advanced/custom_classes.md

```diff
@@ -0,0 +1,177 @@
+# Custom Classes
+
+:::{note}
+Before creating your own custom class to be used by the parameter node wrapper infrastructure, see if [parameter packs](../supported_types.md#parameter-packs) fit your needs.
+
+If you just want to be able to use a custom, reusable widget for a parameter pack, see [Custom Widgets for Parameter Packs](./custom_widgets_for_parameter_packs.md).
+:::
+
+## Using custom classes in a parameterNodeWrapper
+
+This section is meant as a guide to make new types available for use in the `parameterNodeWrapper` and `parameterPack`s. The serialization section must always be done, but the GUI sections are only needed if you want to support GUI connection or creation with your type. New types should be able to be added in extensions (but this is untested).
+
+### Serialization into the underlying vtkMRMLScriptedModuleNode
+
+<details>
+  <summary>Click to expand</summary>
```

Comment:
I'd prefer these not be collapsible; none of the other content collapses like this and it is a bit more difficult to read.

Reply:
Removed the collapsibles.